Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -804,48 +804,210 @@ Another compliant solution is to define the API level to 17 (JELLY_BEAN_MR1) and
+Here is a simple example of how JSON can be written and read using GSON from the GSON User Guide. In this sample, the contents of an instance of the `BagOfPrimitives` is serialized into JSON:
+
+```java
+class BagOfPrimitives {
+  private int value1 = 1;
+  private String value2 = "abc";
+  private transient int value3 = 3;
+  BagOfPrimitives() {
+    // no-args constructor
+  }
+}
+
+// Serialization
+BagOfPrimitives obj = new BagOfPrimitives();
+Gson gson = new Gson();
+String json = gson.toJson(obj);  
+
+// ==> json is {"value1":1,"value2":"abc"}
+
+```
 
-##### ORM
 
-Object-Relational Mapping (ORM) is used to store the contents of an object directly into a database. Libraries like OrmLite<sup>[4]</sup>, SugarORM<sup>[5]</sup>, GreenDAO<sup>[6]</sup> and ActiveAndroid<sup>[7]</sup> use a SQLite database to store the data in. Realm <sup>[8]</sup>, another library, uses its own database to store the contents of a class. 
+##### ORM
+There are libraries that provide the functionality to store the contents of an object directly into a database and then instantiate the objects based on the database content again. This is called Object-Relational Mapping (ORM). 
+There are libraries that use SQLite as a database, such as: OrmLite<sup>[4]</sup>, SugarORM<sup>[5]</sup>, GreenDAO<sup>[6]</sup> and ActiveAndroid<sup>[7]</sup> use a SQLite database to store the data in. Realm <sup>[8]</sup> on the other hand, uses its own database to store the contents of a class. 
 The amount of protection that ORM can provide mostly relies on whether the database is encrypted. See the storage chapter for more details.
+A nice example of ORMLite can be found at <sup>[9]</sup>.
 
 ##### Parcelable
-`Parcelable` is an interface for classes whose instances can be written to and restored from a `Parcel` <sup>[9][10][11]</sup>. A parcel is often used to pack a class as part of a `Bundle` content for an `Intent`. 
+
+`Parcelable` is an interface for classes whose instances can be written to and restored from a `Parcel` <sup>[10][11][12]</sup>. A parcel is often used to pack a class as part of a `Bundle` content for an `Intent`. Here's an example from the Google developer docs that implements `Parcelable`:
+```java
+public class MyParcelable implements Parcelable {
+     private int mData;
+
+     public int describeContents() {
+         return 0;
+     }
+
+     public void writeToParcel(Parcel out, int flags) {
+         out.writeInt(mData);
+     }
+
+     public static final Parcelable.Creator<MyParcelable> CREATOR
+             = new Parcelable.Creator<MyParcelable>() {
+         public MyParcelable createFromParcel(Parcel in) {
+             return new MyParcelable(in);
+         }
+
+         public MyParcelable[] newArray(int size) {
+             return new MyParcelable[size];
+         }
+     };
+     
+     private MyParcelable(Parcel in) {
+         mData = in.readInt();
+     }
+ }
+```
+
+Please note that `Parcelable` is not meant for storing data!
```

Comment:
I think you should be more specific here. Objects can obviously be stored in parcels, so why is it not meant for storing data?